projects/igniteui-angular/src/lib/grids/columns/column.component.ts

```diff
@@ -1855,6 +1868,20 @@ export class IgxColumnComponent implements AfterContentInit, OnDestroy {
         return children;
     }
 
+    /**
+     * Toggles column vibisility and emits the respective event.
+     * @hidden
+     */
+    public toggleVisibility(value?: boolean) {
+        const newValue = value ?? !this.hidden;
+        const eventArgs: IColumnVisibilityChangingEventArgs = { column: this, newValue, cancel: false };
```

Comment:
IMO the value of the args should not be updated here. Because if we cancel the event the column remains in the same state as it were so as the value does.

Reply:
It is expected that the event arguments provide the new value. Also, the name `newValue` suggests the same. Otherwise an event would be emitted that does not contain neither the current nor the old value.

Reply:
We can also use the direct refference of the hidden prop, without changing it's value and have:
{..., value: this.hidden...}
and then once the state of the column is changed pass the new value after it is changed.

It is more logical to me that ing events emits before the any of the operations is done, which means that the value or the state of the current item is still the same otherwise there is no point of canceling it.

Reply:
Which means that we should expose another prop of the arguments which is just the value accompanied with the newValue

Reply:
The more I think about that, the more I see the need of exposing the current and the new value in the ing evt args, but for now let it be like that and if we see an urge about it we are going to expose it.